lib/improver/generate_ancillaries/generate_ancillary.py

```diff
@@ -126,6 +127,7 @@ def process(standard_landmask):
         standard_landmask.data[mask_sea] = False
         mask_land = np.ma.masked_greater(standard_landmask.data, 0.).mask
         standard_landmask.data[mask_land] = True
+        standard_landmask.data = standard_landmask.data.astype(np.int32)
```

Comment:
Is the landmask not required as boolean?  What impact does this have on the output field?

Reply:
Booleans are not supported by netCDF as far as I can tell:
https://www.unidata.ucar.edu/software/netcdf/docs/data_type.html

We certainly get an error from the netCDF4 library when we try to save a cube containing data of type bool.

Reply:
I guess what's confusing me is what datatype was it before this change?